aws/kms/key/key.go

```diff
@@ -286,7 +286,7 @@ func (a *Key) keys() ([]*kms.KeyMetadata, error) {
 	if err != nil {
 		return nil, err
 	}
-	keys := make([]*kms.KeyMetadata, 0, len(entries))
+	keys := make([]*kms.KeyMetadata, len(entries))
```

Comment:
Why remove the default index? It seems like we should be pushing these `resp.KeyMetadata` onto the slice using `append`

Reply:
The `0` in the `make()` function is the length, not the default index. By removing it, the length is the same as the capacity. Instead of using the `append()` function, I'm setting the value using the index in the list response. Using assign vs append is slightly more efficient: [Golang slice append vs assign performance](https://stackoverflow.com/questions/38654729/golang-slice-append-vs-assign-performance)

Reply:
Sounds good, I knew we had seen one odd issue where we were skipping indexes in a different program.